### DISCUSSION TITLE
LangGraph - Does Flowise fit into this?  Or is Flowise an alternative to LangGraph?

### DISCUSSION THREAD

**Discussion by dkindlund** [FlowiseAI/Flowise#1616] [Q&A] [answered]
Hey @HenryHengZJ , this was just announced by the LangChain team -- called LangGraph.  Is this something Flowise could use?  Or is Flowise considered an alternative to LangGraph?  I'm just curious your thoughts on this new project.

Ref:
https://blog.langchain.dev/langgraph/
https://github.com/langchain-ai/langgraph

**Comment by HenryHengZJ** [comment]
Definitely something I would love to bring into Flowise, still looking at whats the best way to do it

**Reply by yos1p** [reply]
Here's just an idea. The power of LangGraph is the workflow / collaboration between Agents. Since Flowise already have Agent nodes, it would be nice that if we have Graph node, that allow several Agent Nodes to be connected in a Graph.

**Comment by dcsan** [comment]
seems this would massively simplify things for you as the project was about normalizing a lot of the langchain interfaces
https://youtu.be/9M8x485j_lU?si=A1b3_3Wt5KMIxhbg&t=187

[wonder how much the JS side is lagging behind?](https://github.com/langchain-ai/langgraphjs/)

presumably you had to do a lot of work to be able to link the different components together as you do esp for your prompt chaining

**Comment by wijjj** [comment]
Dear @HenryHengZJ, is langgraph still on the todo list? Cheers! :)

**Reply by brucechou1983** [reply]
LangGraph support is released.

**Reply by dkindlund** [reply]
Hey @brucechou1983 - this is great!  Do you have any documentation on how to use LangGraph support in Flowise?  Thanks!

**Reply by brucechou1983** [reply]
@dkindlund I didn't see any document yet. i think it basically provides `supervisor` and `worker` nodes in the `agentflow` tab. tried to build a simple workflow but no luck. see this issue https://github.com/FlowiseAI/Flowise/issues/2472

**Reply by brucechou1983** [reply]
in addition, when I use LangGraph, I usually customize many graph states and hierarchies, which I haven’t seen in this release so far. please correct me if i'm wrong.

**Reply by dkindlund** [reply]
Thanks -- I'll keep an eye on #2472.  Once the demos can be reproduced, I'll then mark this issue as resolved at that time.

**Reply by HenryHengZJ** [reply, marked as answer]
@brucechou1983 try disconnect the ChatOpenAI nodes to worker nodes. For more hierarachies and routes, this is coming next iterations. 

@dkindlund for now its just supervisor + worker nodes. Thread - https://x.com/FlowiseAI/status/1792947455094309375

**Reply by dkindlund** [reply]
Hey @HenryHengZJ , any luck on getting workers linked to different models (LLMs)?  I didn't know if this support was released yet.

**Comment by LeonidShamis** [comment]
Does Flowise support the entire LangGraph functionality with all their "primitives" - Nodes, Edges, Conditional Edges, cycles, State?

How are LangGraph's primitives mapped to Flowise's Supervisor, Worker, etc.?

How Conditional Edges can be added to the workflow?

How can cycles be implemented in Flowise?
